### PR TITLE
[Reviewer: Rob] Add in client code

### PR DESCRIPTION
**Pull request by eleanor-merry**
Rob, can you please review the client side changes to support fast Chronos scale down

### CODE REVIEW

File: src/main/timer.cpp [hunk 1]
```diff
@@ -356,6 +360,16 @@ Timer* Timer::from_json(TimerID id, uint64_t replica_hash, std::string json, std
     JSON_PARSE_ERROR(boost::str(boost::format("Failed to parse JSON body, offset: %lu - %s. JSON is: %s") % doc.GetErrorOffset() % doc.GetParseError() % json));
   }
 
+  return from_json_obj(id, replica_hash, error, replicated, doc);  
+}
+Timer* Timer::from_json_obj(TimerID id,
```
Comment (rkd-msw) [inline src/main/timer.cpp]: Missing newline.

File: src/test/test_chronos_internal_connection.cpp [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+  HTTPCode status = _chronos->trigger_move_for_one_server("10.0.0.1");
+  EXPECT_EQ(status, 200);
+}
+
+TEST_F(TestChronosInternalConnection, SendTriggerOneTimer)
+{
+  fakecurl_responses["http://10.42.42.42:80/timers?requesting-node=10.0.0.1;sync-mode=SCALE"] = "{\"Timers\":[{\"TimerID\":4, \"OldReplicas\":[\"10.0.0.2\", \"10.0.0.3\"], \"Timer\": {\"timing\": { \"interval\": 100, \"repeat-for\": 200 }, \"callback\": { \"http\": { \"uri\": \"localhost\", \"opaque\": \"stuff\" }}, \"reliability\": { \"replicas\": [ \"10.0.0.1\", \"10.0.0.3\" ] }}}]}";
+  EXPECT_CALL(*_th, add_timer(_));
+  EXPECT_CALL(*_replicator, replicate_timer_to_node(_, "10.0.0.3"));  // Update
+  EXPECT_CALL(*_replicator, replicate_timer_to_node(_, "10.0.0.2")); // Tombstone
```
Comment (rkd-msw) [inline src/test/test_chronos_internal_connection.cpp]: Should we be testing that one is an update and the other is a tombstone?

File: src/main/chronos_internal_connection.cpp [hunk 3]
```diff
@@ -0,0 +1,349 @@
...
+                                                (rapidjson::Value&)timer_obj);
+
+            if (!timer)
+            {
+              LOG_INFO("Unable to create timer - error: %s", error_str.c_str());
+              continue;
+            }
+            else if (!replicated_timer)
+            {
+              LOG_INFO("Unreplicated timer in response - ignoring");
```
Comment (rkd-msw) [inline src/main/chronos_internal_connection.cpp]: Would we ever expect this to happen? If not, is it a WARNING?

**Comment by rkd-msw** [comment]
Looks good, no major comments.

File: src/test/test_chronos_internal_connection.cpp (hunk 2 above)
Comment (bossmc) [inline src/test/test_chronos_internal_connection.cpp]: We have a "match on a tombstone" matcher somewhere that should make this easy...

**Pull request merged**